clients/src/main/java/org/apache/kafka/clients/producer/internals/ProducerMetadata.java

```diff
@@ -101,8 +127,17 @@ public synchronized void awaitUpdate(final int lastVersion, final long timeoutMs
     }
 
     @Override
-    public synchronized void update(int requestVersion, MetadataResponse response, long now) {
-        super.update(requestVersion, response, now);
+    public synchronized void update(int requestVersion, MetadataResponse response, boolean isPartialUpdate, long nowMs) {
+        super.update(requestVersion, response, isPartialUpdate, nowMs);
+
+        // Remove all topics in the response that are in the new topic set. Note that if an error was encountered for a
+        // new topic's metadata, then any work to resolve the error will include the topic in a full metadata update.
```

Comment:
So we don't retry using another partial request for a failed topic?

Reply:
Correct, per discussion described above.

Reply:
If we see an authorization exception for example, would we retry using a full update straightaway or would it wait 5 minutes for the next refresh interval? Just want to make sure that the delay after these type of errors remains the same as before.

Reply:
If a per-partition error is encountered, then the handling requests a full update, which will retry after the "retry.backoff.ms" has elapsed (default 100 ms).

Reply:
Actually, authorization wouldn't retry right away. Errors deriving from InvalidMetadataException would, however. Handling of both cases is left unchanged.